pkg/antctl/command_definition_test.go

```diff
@@ -439,3 +442,693 @@ func TestCommandDefinitionGenerateExample(t *testing.T) {
+			mode:     "",
+			cd:       &commandDefinition{},
+			expected: false,
+		},
+		{
+			name: "Command for agent defines resource endpoint",
+			mode: "agent",
+			cd: &commandDefinition{
+				agentEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Command for agent defines non-resource endpoint",
+			mode: "agent",
+			cd: &commandDefinition{
+				agentEndpoint: &endpoint{
+					nonResourceEndpoint: &nonResourceEndpoint{
+						path: "/version",
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Command for controller defines resource endpoint",
+			mode: "controller",
+			cd: &commandDefinition{
+				controllerEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Command for controller defines non-resource endpoint",
+			mode: "controller",
+			cd: &commandDefinition{
+				controllerEndpoint: &endpoint{
+					nonResourceEndpoint: &nonResourceEndpoint{
+						path: "/version",
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Command for flow aggregator defines resource endpoint",
+			mode: "flowaggregator",
+			cd: &commandDefinition{
+				flowAggregatorEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Command for flow aggregator defines non-resource endpoint",
+			mode: "flowaggregator",
+			cd: &commandDefinition{
+				flowAggregatorEndpoint: &endpoint{
+					nonResourceEndpoint: &nonResourceEndpoint{
+						path: "/version",
+					},
+				},
+			},
+			expected: false,
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			runtime.Mode = tt.mode
+			actualValue := tt.cd.namespaced()
+			assert.Equal(t, tt.expected, actualValue)
+		})
+	}
+}
+
+func TestAddonTransform(t *testing.T) {
+	tc := []struct {
+		name string
+		cd   *commandDefinition
+		mode string
+	}{
+		{
+			name: "Antctl running against agent mode",
+			mode: "agent",
+			cd: &commandDefinition{
+				agentEndpoint: &endpoint{
+					addonTransform: version.AgentTransform,
+				},
+			},
+		},
+		{
+			name: "Antctl running against controller mode",
+			mode: "controller",
+			cd: &commandDefinition{
+				controllerEndpoint: &endpoint{
+					addonTransform: version.ControllerTransform,
+				},
+			},
+		},
+		{
+			name: "Antctl running against flowaggregator mode",
+			mode: "flowaggregator",
+			cd: &commandDefinition{
+				flowAggregatorEndpoint: &endpoint{
+					addonTransform: version.FlowAggregatorTransform,
+				},
+			},
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			runtime.Mode = tt.mode
+			addonTransform := tt.cd.getAddonTransform()
+			rawResponseData := &Foobar{Foo: "foo"}
+			responseData, err := json.Marshal(rawResponseData)
+			assert.Nil(t, err)
```

Comment:
These lines and line 584 can be out of the test loop. you may just define variables before test execution if they have nothing to change according to different test cases.

Reply:
fixed